.github/workflows/mac-pkg.yml

```diff
@@ -0,0 +1,111 @@
+      PRODUCTSIGN_IDENTITY: ${{ secrets.MACOS_INSTALLER_IDENTITY }}
+      CERTIFICATE_PWD: ${{ secrets.MACOS_CERTIFICATE_PWD }}
+
+      NOTARIZE_TEAM: ${{ secrets.MACOS_NOTARIZATION_TEAM_ID }}
+      NOTARIZE_USERNAME: ${{ secrets.MACOS_NOTARIZATION_APPLE_ID }}
+      NOTARIZE_PASSWORD: ${{ secrets.MACOS_NOTARIZATION_PWD }}
+
+      KEYCHAIN_PWD: ${{ secrets.MACOS_CI_KEYCHAIN_PWD }}
+    steps:
+    - name: Determine Version
+      id: getversion
+      run: |
+        if [ -z "${{ inputs.version }}" ]
+        then
+              VERSION=${{ github.event.release.tag_name }}
+        else
+              VERSION=${{ inputs.version }}
+        fi
+        echo
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+    - name: Check uploads
+      id: check
+      run: |
+        URI="https://github.com/containers/podman/releases/download/${{steps.getversion.outputs.version}}"
+        ARM_FILE="podman-installer-macos-arm64.pkg"
+        AMD_FILE="podman-installer-macos-amd64.pkg"
+
+        status=$(curl -s -o /dev/null -w "%{http_code}" "${URI}/${ARM_FILE}")
+        if [[ "$status" == "404" ]] ; then
+          echo "buildarm=true" >> $GITHUB_OUTPUT
+        else
+          echo "ARM installer already exists, skipping"
+          echo "buildarm=false" >> $GITHUB_OUTPUT
+        fi
+
+        status=$(curl -s -o /dev/null -w "%{http_code}" "${URI}/${AMD_FILE}")
+        if [[ "$status" == "404" ]] ; then
+          echo "buildamd=true" >> $GITHUB_OUTPUT
+        else
+          echo "AMD installer already exists, skipping"
+          echo "buildamd=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Checkout Version
+      if: steps.check.outputs.buildamd == 'true' || steps.check.outputs.buildarm == 'true'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{steps.getversion.outputs.version}}
+    - name: Set up Go
+      if: steps.check.outputs.buildamd == 'true' || steps.check.outputs.buildarm == 'true'
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.18'
+    - name: Create Keychain
+      if: steps.check.outputs.buildamd == 'true' || steps.check.outputs.buildarm == 'true'
+      run: |
+        echo $APPLICATION_CERTIFICATE | base64 --decode -o appcert.p12
+        echo $INSTALLER_CERTIFICATE | base64 --decode -o instcert.p12
+
+        security create-keychain -p "$KEYCHAIN_PWD" build.keychain
+        security default-keychain -s build.keychain
+        security unlock-keychain -p "$KEYCHAIN_PWD" build.keychain
+        security import appcert.p12 -k build.keychain -P "$CERTIFICATE_PWD" -T /usr/bin/codesign
+        security import instcert.p12 -k build.keychain -P "$CERTIFICATE_PWD" -T /usr/bin/productsign
+        security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "$KEYCHAIN_PWD" build.keychain
+
+        xcrun notarytool store-credentials "notarytool-profile" --apple-id "$NOTARIZE_USERNAME" --team-id "$NOTARIZE_TEAM" --password "$NOTARIZE_PASSWORD"
```

Comment:
Github will "try" to mask secret values.  Including in any output/error messages from the tools.  It's not 100% perfect.  e.g. if an error causes a secret to line-wrap, it may not be hidden.  I suggest spending some time going over this with a fine-tooth-comb to make sure no leaks to the console are possible.  Maybe even adding some (very debug frustrating) `&> /dev/null` (assuming mac-bash supports that) where appropriate.

Reply:
Good idea, I'll go through this again and trigger some failures to see if anything pops up.

Reply:
Thanks.  I'm not worried so much about somebody hacking the script in a PR (it won't work).  I think the main threat is from accidental output due to some unforeseen runtime condition.

Reply:
Updated and  /dev/null'd the critical commands